chore(proc): modify renovate configuration

minherz opened the pull request:
## Description

Move configuration under .github/ and upgrade format to json5. aggregate ALL Java dependencies together.
Schedule renovate PRs to be weekly before monday.
Add configs for docker and terraform (just in case) prevent automatic major version upgrades.
Keep special handling for Spring Boot and Micronaut. disable upgrades for Scala and Spark SQL.
Keep previous version of config as renovate.json.legacy

Fixes #8896

lesv reviewed (approved):
LGTM
Note - Oracle Premier support https://www.oracle.com/java/technologies/java-se-support-roadmap.html for Java 11 ended in September, it's probably time to move everything to Java 17.

muncus reviewed (approved):
New file looks good to me.

In order for these to be picked up, we'll need to delete the existing `renovate.json`, as it has a higher precedence than the json5 equivalent.

Git history preserves previous contents of renovate.json, so I dont think we need to keep the legacy file around.

minherz commented:
> In order for these to be picked up, we'll need to delete the existing `renovate.json`, as it has a higher precedence than the json5 equivalent.
> 
> Git history preserves previous contents of renovate.json, so I dont think we need to keep the legacy file around.

I removed "old" file that was initially kept with `.legacy` extension. I added the permlink to the legacy configuration in the comments of `.json5` file
